NodeStepTemplate.yml

```
@@ -0,0 +1,105 @@
+# STEPS LEVEL TEMPLATE:
+# Used to build Node
+# Reusable
+# Used to "hide" the additional variables specific to this SDK which shouldn't be set from a higher level, or
+#   shared from a multi-build pipeline like a publish
+
+parameters:
+- name: ApiSpecSource
+  displayName: ApiSpecSource
+  type: string
+  default: -apiSpecGitUrl https://raw.githubusercontent.com/PlayFab/API_Specs/master/
+- name: CommitMessage
+  displayName: CommitMessage
+  type: string
+  default: Automated build from ADO Pipeline
+- name: GitDestBranch
+  displayName: GitDestBranch
+  type: string
+  default: doNotCommit
+- name: SdkName
+  displayName: SdkName
+  type: string
+  default: NodeSDK
+- name: GitJSetupBranch
+  displayName: GitJSetupBranch
+  type: string
+  default: master
+- name: GitSdkGenBranch
+  displayName: GitSdkGenBranch
+  type: string
+  default: master
+- name: isVersioned
+  displayName: isVersioned
+  type: boolean
+  default: false
+- name: SelfTemplateResource
+  displayName: SelfTemplateResource
+  type: string
+  default: self
+
+steps:
+- checkout: JenkinsSdkSetupScripts
+  clean: true
+  path: s
+- checkout: API_Specs
+  clean: true
+  path: s/API_Specs
+- checkout: SdkGenerator
+  clean: true
+  path: s/SdkGenerator
+- checkout: NodeSdk
+  clean: true
+  submodules: true
+  path: s/sdks/NodeSDK
+  persistCredentials: true
+- bash: |
+    set -e
+    
+    echo alias the ADO variables into local variables
+    ApiSpecSource="${{ parameters.ApiSpecSource }}"
+    CommitMessage="${{ parameters.CommitMessage }}"
+    GitDestBranch="${{ parameters.GitDestBranch }}"
+    SdkName="${{ parameters.SdkName }}"
+    WORKSPACE=$(pwd -W)
+    # Hack attempt to get WORKSPACE into a sub-environment
+    export WORKSPACE="$WORKSPACE"
+    export PF_TEST_TITLE_DATA_JSON="$WORKSPACE//JenkinsSdkSetupScripts//Creds//testTitleData.json"
+    echo === PF_TEST_TITLE_DATA SHOULD BE HERE $PF_TEST_TITLE_DATA_JSON === 
+
+    echo === load util.sh to find msbuild ===
+    if [ -f "util.sh" ]; then
+      . "./util.sh" 2> /dev/null
+    elif [ ! -z "$WORKSPACE" ]; then
+      . "$WORKSPACE/JenkinsSdkSetupScripts/JenkinsScripts/Pipeline/util.sh" 2> /dev/null
+    fi
+    
+    Find2019MsBuild || Find2017MsBuild
+    
+    . "$WORKSPACE/JenkinsSdkSetupScripts/JenkinsScripts/Pipeline/sdkUtil.sh" 2> /dev/null
```

```suggestion
    . "$WORKSPACE/JenkinsSdkSetupScripts/JenkinsScripts/Pipeline/sdkUtil.sh"
```
I've been trying to remove these:  2> /dev/null